umpost/conversion_driver_esm1p5.py

```diff
@@ -154,6 +156,8 @@ def convert_fields_file_list(fields_file_paths, nc_write_dir):
         try:
             um2netcdf4.process(fields_file_path, nc_write_path, ARG_VALS)
             succeeded.append((fields_file_path, nc_write_path))
+            if delete_ff:
+                os.remove(fields_file_path)
```

Comment:
Adding deletion code here adds an additional responsibility for the _conversion_ function. The deletion flag also touches a couple of the workflow function signatures.

I'd recommend removing the `delete_ff` handlers from the code &  using `__main__` for deletion like:
```Python
if args.delete_ff:
    # TODO: probably test that the conversion was OK
    # TODO: only delete successes?
    for path in successful_paths:
        os.remove(path)
```
I think it might be best as the final conversion driver step. This has the effect of running the conversion workflow first. In the event the conversions fail, deletion won't occur during that step & potentially leave ESM outputs in an incomplete state.

Unit testing deletion can also be skipped if there's only 2-3 lines of simple code (we visually confirm it's correct).

Reply:
Sweet, that sounds like a good idea. I've added this in, and added in an extra check to ensure that none of the successful conversion paths also appear in the failed conversions:

https://github.com/ACCESS-NRI/um2nc-standalone/blob/b86c0b2820610e0b6fb6ab04a708dfa79488b1b4/umpost/conversion_driver_esm1p5.py#L310-L322

Let me know if this is looking ok!

Reply:
Just realised I forgot to add the message to the PostProcessingError... Added it in.

Reply:
And just realised I forgot to modify the tests... on a roll today. Will fix them now.